### PR TITLE
tools: For ec pools list objects in all shards if the pgid doesn't sp…

### PR DESCRIPTION
**Pull request by dzafman**
…ecify

Signed-off-by: David Zafman <dzafman@redhat.com>

### CODE REVIEW

File: src/tools/ceph_objectstore_tool.cc [hunk 1]
```diff
@@ -434,7 +434,54 @@ int _action_on_all_objects_in_pg(ObjectStore *store, coll_t coll, action_on_obje
...
+  }
+  pgid.parse(pgidstr.c_str());
+  for (vector<coll_t>::iterator i = candidates.begin();
+       i != candidates.end();
+       ++i) {
+    spg_t cand_pgid;
+    snapid_t snap;
+    if (!i->is_pg(cand_pgid, snap))
+      continue;
+    if (snap != CEPH_NOSNAP)
```
Comment (dachary) [inline src/tools/ceph_objectstore_tool.cc]: is_pg may set snap ?

Comment (dachary) [inline src/tools/ceph_objectstore_tool.cc]: I guess it does. I got confused by the **is_pg** function name that looks like it's a predicate with no side effect.

**Comment by dachary** [comment]
it would be nice to have a test somewhere verifying this. Maybe you already have something in ceph-qa-suite ?

File: src/tools/ceph_objectstore_tool.cc (hunk 1 above)
Comment (dzafman) [inline src/tools/ceph_objectstore_tool.cc]: I believe at one time snapshots use to be contained in the collection names.  That is why you can see 1.3_head as the collection name for pg 1.3 in filestore.  However, now the snapshots are handled by the objectstore itself within the collections.  I'm not sure when you'd ever see anything other than CEPH_NOSNAP (_head) in an spg_t unless an old filestore had that kind of collection still laying around.  We need to talk to Sam to confirm possibly getting rid of dead code and do some cleanup.

**Comment by dzafman** [comment]
I don't have a test for this new feature.  It isn't critical functionality, so I wasn't going to bother.

**Pull request merged**